In [21]:
import numpy as np
import pandas as pd
from math import factorial

In [22]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [6]:
salary_list = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [7]:
salary = np.array(salary_list)

In [10]:
# среднее арифметическое = сумма / количество
middle = sum(salary)/len(salary)
middle

65.3

In [16]:
# среднее квадратичное отклонение показывает, насколько далеко наблюдения могут быть «разбросаны» относительно среднего значения
salary_std = 0
for i in salary:
    salary_std += (i - middle)**2
salary_std = np.sqrt(salary_std / len(salary))
salary_std

30.823854398825596

In [17]:
# смещенная оценка дисперсии для данной выборки равна среднему квадратичному отклонению, возведенному в квадрат:
salary_variance = salary_std**2
salary_variance

950.11

In [19]:
# отличие несмещенной оценки дисперсии от смещенной в том, что мы делим сумму квадратов разностей всех величин со средним значением не на 𝑛, а на 𝑛−1.
salary_std_2 = 0
for i in salary:
    salary_std_2 += (i - middle)**2
salary_std_2 = np.sqrt(salary_std_2 / (len(salary) - 1))
salary_std_2

31.624607341019814

In [20]:
salary_variance = salary_std_2**2
salary_variance

1000.1157894736843

**Ответ**: среднее арифметическое - 65.3, среднее квадратичное отклонение - 30.8239, смещенная оценка дисперсии - 950.11, несмещенная оценка дисперсии - 1000.16

### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [28]:
# У нас на выходе6 мячей из которых 3 белые, посмотрим сколько комбинаций такого расклада возможно:
combinations(6, 3)

20

In [33]:
# 20 комбинаций - это достаточно много для перебора каждой. Посмотрим каждую детально.

# 2б-1б3ч(2 белых и 1 белая и 3 черных)
print(f"Количество комбинаций для набора 2б-1б3ч: {combinations(2, 2)*combinations(4, 1)}")

Количество комбинаций для набора 2б-1б3ч: 4


In [34]:
print(f"Количество комбинаций для набора 1б1ч-2б2ч: {combinations(2, 1)*combinations(4, 2)}")

Количество комбинаций для набора 1б1ч-2б2ч: 12


In [35]:
print(f"Количество комбинаций для набора 3ч-3б1ч: {combinations(2, 0)*combinations(4, 3)}")

Количество комбинаций для набора 3ч-3б1ч: 4


In [42]:
# Как мы видим у нас получилось 4, 12 и 4, что в сумме как раз и есть 20. По этому упростим себе задачу:
# 2б-1б3ч
first_case = ((5/8)*(4/7))*((5/12)*(7/11)*(6/10)*(5/9))*4
first_case

0.12626262626262624

In [43]:
# 1б1ч-2б2ч
second_case = ((5/8)*(3/7))*((5/12)*(4/11)*(7/10)*(6/9))*12
second_case

0.22727272727272724

In [44]:
# 3ч-3б1ч
third_case = ((3/8)*(2/7))*((5/12)*(4/11)*(3/10)*(7/9))*4
third_case

0.01515151515151515

Таким образом вероятности посчитаны для каждого случая, объединим их:

In [45]:
P = first_case + second_case + third_case
P

0.3686868686868686

**Ответ**: Вероятность, что три мяча белые: 0.36(86) или 36.9%.

### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

In [100]:
p1 = 0.9
p2 = 0.8
p3 = 0.6
q1 = 1 - p1
q2 = 1 - p2
q3 = 1 - p3

In [101]:
p1*q2*q3

0.072

In [102]:
q1*p2*q3

0.031999999999999994

In [103]:
q1*q2*p3

0.011999999999999993

**Ответ**: Вероятность, первый спортсмен с первого раза попал в цель: 0.072 или 7.2%, что второй 0.0319 или 3.2%, третий 0.0119 или 1.19%.

### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [87]:
# вероятность того что случайный студент из факультета A
qa = 0.25
# вероятность того что случайный студент из факультета B
qb = 0.25
# вероятность того что случайный студент из факультета B
qc = 0.5

In [88]:
pa = 0.8
pb = 0.7
pc = 0.9

In [89]:
# доля сдавших студентов от общего количества поступивших
pt = qa*pa+qb*pb+qc*pc
print(f"Доля сдавщих студентов от общего количества поступивших - {pt}")

Доля сдавщих студентов от общего количества поступивших - 0.825


In [91]:
p = qa*pa/pt
p

0.24242424242424246

In [92]:
p = qb*pb/pt
p

0.21212121212121213

In [93]:
p = qc*pc/pt
p

0.5454545454545455

**Ответ**: Вероятность, что факультет А = 24.24%, B = 21,21%, C = 54,54%.

### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [79]:
# Вероятности выйти из строя для каждой детали
p1 = 0.1
p2 = 0.2
p3 = 0.25

In [95]:
q1 = 1 - p1
q2 = 1 - p2
q3 = 1 - p3

In [94]:
# вероятность того, что в первый месяц выйдут из строя: а). все детали
p1*p2*p3

0.005000000000000001

In [97]:
# Какова вероятность того, что в первый месяц выйдут из строя: б). только две детали
p1*p2*q3 + p1*q2*p3 + q1*p2*p3

0.08000000000000002

In [96]:
# вероятность того, что в первый месяц выйдут из строя: в). хотя бы одна деталь
1 - q1*q2*q3

0.45999999999999996

In [98]:
# вероятность того, что в первый месяц выйдут из строя: г). от одной до двух деталей?
p1*q2*q3 + q1*q2*p3 + q1*p2*q3 + (p1*p2*q3 + p1*q2*p3 + q1*p2*p3)

0.45500000000000007

**Ответ**: a) 0.5% б) 8% в) 45.99% г) 45.5%